elastic/kibana
# [Uptime] Use URL from the /allowed route in the action button to signup for Synthetics beta

## Summary

As discussed earlier today, this PR updates the monitor management screen so that the monitor management screen can point users to the Google form to signup for beta.

It uses whatever value is returned by the service in the `signupUrl` field. That URL should already include the `queryParam` with the user's `accountId` to pre-fill in the form.


## How to test this PR

1. Checkout the appropriate branch on the service side (won't link here as it's a private repo).
2. Run the service as described on the related service PR. Make sure to add the environment variables with the URL to the google form and the correct param. You can find more information about this form [here](https://github.com/elastic/kibana/issues/128594).
    Make sure to run the service in a way which makes it return `false` for your user's `allowed` flag. This can be done by simply hard-coding `false` into the `/allowed` route's response.
    The variables you need to set are:
    * `SIGNUPURL` with the form's URL (https://docs.google.com/forms/d/e/1FAIpQLSfWPWhpGmKzE8KteuRywVRqqKEhahG-494q8tIldZMp3xo2SQ/viewform)
    * `SIGNUPACCOUNTIDFIELD` with the form's AccountID field ID (i.e. `entry.777059315` for the form above)
3. Start Kibana and access the monitor management page.
4. You should see that the button to request access to the service should be enabled.
5. Clicking that button should take you to your Google Form's URL and pass it the correct query param (also specified as an environment variable when running the service). The Google Form should contain the field with the accountId pre-filled.

![Screenshot 2022-03-29 at 18 58 04](https://user-images.githubusercontent.com/6868147/160676117-51833fdb-9c74-4c9a-8fd8-ab857478af1d.png)

After testing the button is _enabled_, test the opposite: make sure your user is _allowed_ to access the service, and ensure you can still add monitors and see everything else under monitor management.

### Checklist

Delete any items that are not applicable to this PR.



### For maintainers



## Release Note

Add button which allows users to signup for the Synthetics service public beta.
